src/awkward/contents/bytemaskedarray.py

```diff
@@ -290,6 +304,7 @@ def _getitem_at(self, where):
 
     def _getitem_range(self, where):
         if not self._backend.nplike.known_shape:
+            self._touch_shape(recursive=False)
```

Comment:
No (it's not a performance thing), it's because it's already touched, since this is not a specialized type-tracer code path. In the case of `ByteMaskedArray.length` (and any other `Content.length`, but taking this as an example),

https://github.com/scikit-hep/awkward/blob/b695e2fe78fbdb0a810a06a92e9fdd3aa3931d12/src/awkward/contents/bytemaskedarray.py#L178-L180

calls

https://github.com/scikit-hep/awkward/blob/b695e2fe78fbdb0a810a06a92e9fdd3aa3931d12/src/awkward/index.py#L123-L125

which calls

https://github.com/scikit-hep/awkward/blob/b695e2fe78fbdb0a810a06a92e9fdd3aa3931d12/src/awkward/_typetracer.py#L372-L375

so the shape gets touched.